### ISSUE TITLE
XSS & CSRF vulnerabilities - CVE-2017-14506

### ISSUE THREAD

**Issue by Quitten** [geminabox/geminabox#278]
Please see the following blogpost. it is possible to perform XSS (via malicious gem) & CSRF attacks on geminabox systems. please address this issue ASAP.
Full details:
http://baraktawily.blogspot.co.il/2017/09/gem-in-box-xss-vulenrability-cve-2017.html

**Comment by reggieb** [comment]
I've added a comment to that blog post:

> You have spotted a potential problem with the web interface to Geminabox - in that it will allow a malicious gem to modify the web page and inject JavaScript code into the page. However, I fail to see how that will "delete any gems on the server" or "won't let users use the geminabox anymore."

**This does need to be fixed** - but you're description of the potential risks is out of proportion to the actual risk.

**Comment by Quitten** [comment]
Via XSS attack, you can send requests to the server on behalf of who is visiting the website,
the malicious javascript runs under the geminabox host, so whenever one visit the webisite, the malicious script will enumerate the uploaded gems, sends the relevant DELETE requests to the server, and they will be deleted.
In addition, it is possible to cause DoS to the web application by injecting malicious javascript that crash the browser while clients accessing geminabox web app or just redirect the user's from the index page so they won't be able to use it.

**Comment by sonots** [comment]
0.13.6 is released.

**Comment by barakt11** [comment]
Yes you are right i tried to see if there is email i can contact with you guys and didnt find nothing.
And what about CSRF protection? you can still enforce users to upload malicious gem into them gems store

**Comment by sonots** [comment]
You could find email from https://rubygems.org/gems/geminabox like https://rubygems.org/profiles/sonots.

I could not see CSRF vulnerability from your blog post.

Discussions should be do continued privately.

**Comment by Quitten** [comment]
Oh right, my bad i didn't check the rubygems page, will continue privately.
